# M1L5 Pandas Part 2 Data Challenge:  EDA

## Scenario

We'll be working with a real-world dataset from the NYC Open Data portal, focusing on the leading causes of death in New York City (same as data challenge 4). This dataset provides valuable insights into public health trends and disparities. Understanding this data is crucial for community advocacy and policy-making.

For more information about the data (which is highly recommended) here is the [Link to the Data](https://data.cityofnewyork.us/Health/New-York-City-Leading-Causes-of-Death/jb7j-dtam/about_data)

## Objectives 
- Group and Aggregate Data
- Create a contingency table with `crosstab()`
- Check for duplicated data (remember not all duplicated data needs to be dropped)

**Let's get started!**

### Step 1:  Import Pandas & Numpy

In [1]:
# Import Pandas & Numpy
import pandas as pd
import numpy as np

### Step 2: Load the dataset (csv file stored in the data folder) into a Pandas DataFrame. The file is called:  `nyc_causeofdeath.csv`


In [2]:
df = pd.read_csv("/Users/Marcy_Student/desktop/marcy/DA2025_Lectures_Kevin/data/New_York_City_Leading_Causes_of_Death_20250605.csv",delimiter=",")

### Step 3: Check the information of the data (column names, data types, size, etc.)


In [ ]:
df.dtypes

7658

### Step 4:  We need to change Deaths from an object to an integer so that we can do some MATH (you will learn this officially later) for now just run the cell below.


In [9]:
#Run this cell without changes 
df['Deaths'] = df['Deaths'].replace('.', np.nan)
df['Deaths'] = pd.to_numeric(df['Deaths'])
df.keys()

Index(['Year', 'Leading Cause', 'Sex', 'Race Ethnicity', 'Deaths',
       'Death Rate', 'Age Adjusted Death Rate'],
      dtype='object')

### Step 5:  Create code to get the sum of deaths by Sex -- what Sex has the most deaths based on this data (add a comment in the cell with your answer)

In [12]:
deaths_by_sex = df.groupby('Sex')['Deaths'].sum()
print(deaths_by_sex)

Sex
F    217071.0
M    207927.0
Name: Deaths, dtype: float64


Females have more deaths with 217,071 and Males 207,927

### Step 6:  Now create a contingency table (using `crosstab()`) of the Leading Cause of Death by Sex -- put a comment in the cell of a takeaway from the output 

In [14]:
cause_by_sex = pd.crosstab(df['Leading Cause'],df['Sex'] )
cause_by_sex

Sex                                                  F   M
Leading Cause                                             
Accidents Except Drug Posioning (V01-X39, X43, ...  35  45
All Other Causes                                    48  48
Alzheimer's Disease (G30)                           31   1
Aortic Aneurysm and Dissection (I71)                 2   1
Assault (Homicide: Y87.1, X85-Y09)                   3  17
Atherosclerosis (I70)                                3   0
Cerebrovascular Disease (Stroke: I60-I69)           48  42
Certain Conditions originating in the Perinatal...  13  13
Chronic Liver Disease and Cirrhosis (K70, K73)       8  21
Chronic Lower Respiratory Diseases (J40-J47)        45  43
Congenital Malformations, Deformations, and Chr...   6   2
Diabetes Mellitus (E10-E14)                         47  45
Diseases of Heart (I00-I09, I11, I13, I20-I51)      48  48
Essential Hypertension and Renal Diseases (I10,...  46  29
Human Immunodeficiency Virus Disease (HIV: B20-...  17  26
Influenza (Flu) and Pneumonia (J09-J18)             48  48
Insitu or Benign / Uncertain Neoplasms (D00-D48)     5   0
Intentional Self-Harm (Suicide: X60-X84, Y87.0)     12  26
Malignant Neoplasms (Cancer: C00-C97)               48  48
Mental and Behavioral Disorders due to Accident...  12  27
Mental and Behavioral Disorders due to Use of A...   0   2
Nephritis, Nephrotic Syndrome and Nephrisis (N0...  14   3
Parkinson's Disease (G20)                            1   0
Septicemia (A40-A41)                                11   2
Tuberculosis (A16-A19)                               1   0
Viral Hepatitis (B15-B19)                            2   3

Alzheimer's Disease (G30) affects women way more than males (31:1).

### Step 7:  Are there any duplicate records in this dataset?  Code it below and add a comment with your answer

In [ ]:
keys = list(df.keys())
for i in keys:
    print(i,":", df[i].duplicated().sum())
    

Year : 1086
Leading Cause : 1068
Sex : 1092
Race Ethnicity : 1088
Deaths : 629
Death Rate : 652
Age Adjusted Death Rate : 667


## Above and Beyond (AAB)  -- OPTIONAL

### Question 1:  What year had the most deaths?

In [ ]:
df.groupby('Year')['Deaths'].sum()

Year
2007    141
2008    136
2009    135
2010    138
2011    141
2012    134
2013    133
2014    136
dtype: int64

2007 and 2011 had the most deaths

### Question 2:  Change the 'Death Rate' column to a float.  Why would you want to do this?

In [ ]:
df['Death Rate'] = df['Death Rate'].astype(float)


dtype('float64')

You might want to convert to float because a) its a decimal value and b) in order to do calculations on the values. 